python_modules/dagster/dagster/_core/definitions/schedule_definition.py

```diff
@@ -512,7 +512,13 @@ def with_updated_job(self, new_job: ExecutableDefinition) -> "ScheduleDefinition
             description=self.description,
             job=new_job,
             default_status=self.default_status,
+            run_config=None,
+            run_config_fn=self._run_config_fn,
+            tags=self._tags,
+            tags_fn=self._tags_fn,
+            environment_vars=self._environment_vars,
             required_resource_keys=self.required_resource_keys,
+            should_execute=None,
```

Comment:
please add comments on the `None` arguments so that we understand fully why this is not a bug